manifests/source.pp

```diff
@@ -79,8 +79,8 @@
   $_before = Apt::Setting["list-${title}"]
 
   if !$release {
-    if $facts['os']['distro']['codename'] {
-      $_release = $facts['os']['distro']['codename']
+    if fact('os.distro.codename') {
+      $_release = fact('os.distro.codename')
```

Comment:
Very strictly speaking, this line doesn't need to be modified since at this point you know it's available. But it works too.